### ISSUE TITLE
Add format transform to parse json strings, csv etc

### ISSUE THREAD

**Issue by PhML** [vega/vega#2779]
We can specify the format of data when we load external data in a vega spec. But I am in a case where I connect to a service that returns this kind of data:

```json
{
  "set1": "{\"subset1\": [1, 3, 5]}",
  "set2": "{\"subset2\": [0,2,4]}"
}
```

Where `set1` value is a string. I'd like to access `subset1` values but it is not (yet?) possible. A solution would be to add a *json* transform so we could do:

```json
{
  "transform": [
    { "type": "parse", "field": "set1", "format": "json" },
    { "type": "parse", "field": "set2", "format": "json" }
    ]
 }
```

In this case its a json sting in a json structure but we can imaging having a CSV file with json string values (yes I have already seen this kind of stuff!).

Please:

- [x] Check for duplicate issues. Please file separate requests as separate issues on GitHub.
- [x] Clearly describe the goal of the feature request. Why is it needed?
- [x] Describe an envisioned solution. What do you want the new behavior to be?
- [ ] If applicable, share mockup images or links to examples illustrating the desired output.
- [x] If applicable, share JSON spec designs for how the behavior would be made available in Vega.

**Comment by domoritz** [comment]
Yeah, I don't think we support this use case and likely won't as (imho) the data source should be fixed. For now, you can fetch the data yourself, parse it, and then add it to Vega with https://vega.github.io/vega/docs/api/view/#view_insert. Is that an acceptable solution?

**Comment by PhML** [comment]
In fact, I write a [Grafana](https://grafana.com/) plugin where user can fill a Vega spec to visualize data from a PostgreSQL datasource (it could be another datasource).

Generally I retrieve simple types of data and everything is OK. But in a specific use case I retrieve a Json string from the database that I need to parse.

As it is specific I’d rather like not parse it before to add it to Vega as it would break normal use cases. I also rather like to let user decide what to do with his data in the spec instead of manipulate it in between.

To me it is really a transform that could be done in Vega itself: it is not a complex task and (I think that) all the bricks are already there. There are more complex tasks possible to do with transforms. Here it is just about *reading* more versatile data representation. It also could be used in other contexts without the need to add a conversion layer between the data source and Vega.

That's my opinion, I am not a Vega dev, I can't see all the implications so the decision is up to you.

**Comment by hydrosquall** [comment]
I've addressed this use case by adding a parseJson function via the [extensibility](https://vega.github.io/vega/docs/api/extensibility/) api. Then, the strings can be parsed into a field as objects using a formula.

We could either document this solution, or add jsonParse to the list of builtin expressions . Would that work for you @PhML ?
